polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -674,18 +574,13 @@ $loading-panel-height: 53px;
   left: 0;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   z-index: var(--pc-index-table-sticky-cell);
-  padding: var(--p-space-2) 0;
+  padding: var(--p-space-1_5-experimental) 0;
   vertical-align: middle;
-
-  #{$se23} & {
-    padding: var(--p-space-1_5-experimental) 0;
-  }
 }
 
 .TableCell-first + .TableCell {
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   left: var(--pc-checkbox-offset);
-  padding-left: 0;
```

Comment:
I removed this, because it was causing unwanted layout shift on sticky
Tested this quite a bit, whatever was causing the extra padding no longer exists so I removed this to resolve the prior layout shift bug.